### PR TITLE
docs: Release notes for BLS upgrade

### PR DESCRIPTION
**Pull request by ogabrielides**
## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: doc/release-notes-5021.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+`MNHFTx` will support a new `version` value:
+
+| Version | Version Description                            |
+|---------|------------------------------------------------|
+| 1       | Serialisation of `sig` using legacy BLS scheme |
+| 2       | Serialisation of `sig` using basic BLS scheme  |
+
+BLS enforced scheme
+--------
+Once v19 hard fork is activated all remaining messages containing BLS public keys or signatures will serialise them using the new basic BLS scheme.
```
Comment (thephez) [inline doc/release-notes-5021.md:67]:
```suggestion
Once the v19 hard fork is activated, all remaining messages containing BLS public keys or signatures will serialise them using the new basic BLS scheme.
```

File: doc/release-notes-5021.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+List of affected messages:
+`dsq`, `dstx`, `mnauth`, `govobj`, `govobjvote`, `qrinfo`, `qsigshare`, `qsigrec`, `isdlock`, `clsig`, all DKG messages (`qfcommit`, `qcontrib`, `qcomplaint`, `qjustify`, `qpcommit`);
```
Comment (thephez) [inline doc/release-notes-5021.md:70]:
```suggestion
`dsq`, `dstx`, `mnauth`, `govobj`, `govobjvote`, `qrinfo`, `qsigshare`, `qsigrec`, `isdlock`, `clsig`, and all DKG messages (`qfcommit`, `qcontrib`, `qcomplaint`, `qjustify`, `qpcommit`).
```

**Review by thephez — CHANGES_REQUESTED** [review]
Some small suggestions...

File: doc/release-notes-5021.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+Added RPCs
+--------
+
+- Once the v19 hard fork is activated, `protx register`, `protx register_fund`, and `protx register_prepare` RPCs will decode BLS operator public keys using the new basic BLS scheme. In order to support BLS public keys encoded in the legacy BLS scheme, `protx register_legacy`, `protx register_fund_legacy`, and `protx register_prepare_legacy` were added.
+
+Other changes
+--------
+
+`qcommit`
```
[the remark this answers is not in the document]
Comment (ogabrielides) [reply doc/release-notes-5021.md:9]: I am referring to the final commitment mined on chain. So yes `qfcommit` is the correct one.

File: doc/release-notes-5021.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+`qfcommit`
+--------
+
+Once the v19 hard fork is activated, `quorumPublicKey` will be serialised using the basic BLS scheme.
+To support syncing of older blocks containing the transactions using the legacy BLS scheme, the `version` field indicates which scheme to use for serialisation of `quorumPublicKey`.
+
+| Version | Version Description                                    | Includes `quorumIndex` field |
+|---------|--------------------------------------------------------|------------------------------|
+| 1       | Non-rotated qcommit serialised using legacy BLS scheme | No                           |
```
Comment (thephez) [inline doc/release-notes-5021.md:17]:
```suggestion
| 1       | Non-rotated qfcommit serialised using legacy BLS scheme | No                           |
```

File: doc/release-notes-5021.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+| 2       | Rotated qcommit serialised using legacy BLS scheme     | Yes                          |
```
Comment (thephez) [inline doc/release-notes-5021.md:18]:
```suggestion
| 2       | Rotated qfcommit serialised using legacy BLS scheme     | Yes                          |
```

File: doc/release-notes-5021.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+| 3       | Non-rotated qcommit serialised using basic BLS scheme  | No                           |
```
Comment (thephez) [inline doc/release-notes-5021.md:19]:
```suggestion
| 3       | Non-rotated qfcommit serialised using basic BLS scheme  | No                           |
```

File: doc/release-notes-5021.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+| 4       | Rotated qcommit serialised using basic BLS scheme      | Yes                          |
```
Comment (thephez) [inline doc/release-notes-5021.md:20]:
```suggestion
| 4       | Rotated qfcommit serialised using basic BLS scheme      | Yes                          |
```

**Review by thephez — COMMENTED** [review]
Maybe these too?

**Review by thephez — APPROVED** [review]
ACK

**Comment by ogabrielides** [comment]
@PastaPastaPasta @UdjinM6 Could you review this please?

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by UdjinM6 — APPROVED** [review]
utACK

**Pull request merged**